.github/workflows/ci.yaml

```diff
@@ -8,74 +8,94 @@ on:
       - uses: actions/checkout@v3
-      - uses: asdf-vm/actions/install@v2
-      - run: >
-          find . -name '*.dart'
-          ! -name '*.freezed.dart'
-          ! -name '*.g.dart'
-          ! -name '*.mocks.dart'
-          ! -path '*/l10n/*.dart'
-          ! -name '*.pb*.dart'
-          ! -path '*/.*/*'
-          | xargs dart format --set-exit-if-changed
+      - name: Restore repository cache
+        uses: actions/cache@v3
+        with:
+          path: .
+          key: ${{ runner.os }}-repo-${{ github.sha }}-generated
+          restore-keys: |
+            ${{ runner.os }}-repo-
+      - uses: Atsumi3/actions-setup-fvm@0.0.3
+      - uses: bluefireteam/melos-action@v2
+      - run: melos format:exclude
 
   test:
-    needs: analyze
     runs-on: ubuntu-22.04
+    needs: setup
     steps:
       - uses: actions/checkout@v3
-      - uses: asdf-vm/actions/install@v2
-      - uses: actions/cache@v3
+      - name: Restore repository cache
+        uses: actions/cache@v3
         with:
-          path: |
-            ./**/*.freezed.dart
-            ./**/*.g.dart
-            ./**/*.mocks.dart
-          key: build_runner-${{ github.sha }}
+          path: .
+          key: ${{ runner.os }}-repo-${{ github.sha }}-generated
+          restore-keys: |
+            ${{ runner.os }}-repo-
+      - uses: Atsumi3/actions-setup-fvm@0.0.3
+      - uses: bluefireteam/melos-action@v2
       - run: sudo apt update && sudo apt install -y lcov
-      - run: flutter test --coverage
+      - run: melos coverage
       - uses: codecov/codecov-action@v3
         with:
-          token: ${{secrets.CODECOV_TOKEN}}
+          token: ${{ secrets.CODECOV_TOKEN }}
 
   security-center:
-    needs: analyze
     runs-on: ubuntu-22.04
+    needs: setup
     steps:
       - uses: actions/checkout@v3
-      - uses: asdf-vm/actions/install@v2
-      - uses: actions/cache@v3
+      - name: Restore repository cache
+        uses: actions/cache@v3
         with:
-          path: |
-            ./**/*.freezed.dart
-            ./**/*.g.dart
-            ./**/*.mocks.dart
-          key: build_runner-${{ github.sha }}
-      - run: |
-          sudo apt update
-          sudo apt install -y clang cmake libglib2.0-dev libgtk-3-dev liblzma-dev ninja-build pkg-config polkitd xvfb
-          sudo snap refresh snapd --channel=latest/edge/prompting
-          sudo snap install --edge prompting-client
+          path: .
+          key: ${{ runner.os }}-repo-${{ github.sha }}-generated
+          restore-keys: |
+            ${{ runner.os }}-repo-
+      - uses: Atsumi3/actions-setup-fvm@0.0.3
+      - uses: bluefireteam/melos-action@v2
+      - name: Install dependencies
+        run: sudo apt update && sudo apt install -y clang cmake libglib2.0-dev libgtk-3-dev liblzma-dev ninja-build pkg-config polkitd xvfb
+      - name: Setup apparmor prompting
+        run: |
+          sudo snap refresh snapd --channel=latest/edge
+          sudo snap install prompting-client
```

Comment:
do we potentially want to do a matrix here over the edge and stable channels of the prompting client?

Reply:
The client itself isn't being used directly (I don't think) but if there are any future changes like the ones in snapd that require the client to have certain interfaces present it might help catch any strange interactions that we might otherwise miss.

Reply:
right, the only reason it's being installed is so that we can actually enable the feature, which now requires a client to be present. does it make sense to run those checks here in the CI for the security center, though?